### PR TITLE
MWPW-128687 Update auth handling in Floodgate AdobeIO app

### PR DESCRIPTION
**Pull request by raga-adbe-gh**
Changes in this are related to moving the auth from user supplied access token to app based. 
Process mentioned for app based authentication mentioned in https://git.corp.adobe.com/jsingler/sp-auth  has been followed.
As part of this few new env entries have been added. For local testing either these can be added or Base64 encoded current accesstoken can be added to TEST_ACCESS_TOKEN can be set in env with base64 

Sample Base64 in terminal $ echo -n '<accesstoken>' | base64

Resolves: MWPW-128687

Few Test Runs Screenshots
**Few of previous runs**
image
**Sample Log**
image

### CODE REVIEW

File: actions/copy/copy.js [hunk 1]
```diff
@@ -21,26 +21,29 @@ const { PROJECT_STATUS } = require('../project');
 const {
     getAioLogger, updateStatusToStateLib, COPY_ACTION, getStatusFromStateLib
 } = require('../utils');
+const appConfig = require('../appConfig');
```
Comment (sirivuri) [inline actions/copy/copy.js:24]: @raga-adbe-gh should this be configMap??

**Review by sukamat — APPROVED** [review]

**Review by arshadparwaiz — APPROVED** [review]

File: actions/appConfig.js [hunk 2]
```diff
@@ -1,35 +1,76 @@
...
+        this.configMap.rootFolder = params.rootFolder;
+        this.configMap.fgRootFolder = params.fgRootFolder;
+        this.configMap.clientId = params.clientId;
+        this.configMap.tenantId = params.tenantId;
+        this.configMap.driveId = params.driveId;
+        this.configMap.certPassword = params.certPassword;
+        this.configMap.certKey = params.certKey;
+        this.configMap.certThumbprint = params.certThumbprint;
+        this.configMap.testAccessToken = params.testAccessToken;
+        this.configMap.skipInProg = (params.skipInProgressCheck || '').toLowerCase() === 'true';
```
Comment (sukamat) [inline actions/appConfig.js:36]: This is nice for local testing :)

File: actions/appConfig.js [hunk 3]
```diff
@@ -1,35 +1,76 @@
...
+        this.extractPrivateKey();
+        // Test token in base64
+        if (this.configMap.testAccessToken) {
+            this.configMap.testAccessToken = Buffer.from(this.configMap.testAccessToken, 'base64');
```
Comment (sukamat) [inline actions/appConfig.js]: As discussed, let's not have this option. Let all FG actions use the app based token.

File: actions/sharepoint.js [hunk 4]
```diff
@@ -308,6 +317,20 @@ async function fetchWithRetry(apiUrl, options, retryCounts) {
...
+function logHeaders(response) {
+    if (!LOG_RESP_HEADER) return;
+    const logger = getAioLogger();
+    const headers = {};
+    response.headers.forEach((value, name) => {
+        headers[name] = value;
+    });
+    const hdrStr = JSON.stringify(headers);
+    const logStr = `Status is ${response.status} with headers ${hdrStr}`;
+    // logStr  = logStr + new Error().stack.replace('Error','');
```
Comment (sukamat) [inline actions/sharepoint.js]: clean up commented code

File: app.config.yaml [hunk 5]
```diff
@@ -9,6 +9,18 @@ application:
           fgSite: $FG_SITE
           fgClientId: $FG_CLIENT_ID
           fgAuthority: $FG_AUTHORITY
+          shareUrl: $SHARE_URL
+          fgShareUrl: $FG_SHARE_URL
+          rootFolder: $ROOT_FOLDER
+          fgRootFolder: $FG_ROOT_FOLDER
+          skipInProgressCheck: $SKIP_INPROGRESS_CHECK
+          testAccessToken: $TEST_ACCESS_TOKEN
```
Comment (sukamat) [inline app.config.yaml]: `testAccessToken: $TEST_ACCESS_TOKEN` - this can be removed

**Review by sukamat — COMMENTED** [review]
@raga-adbe-gh Looks good! As discussed offline, let's make those minor changes before merging this PR.

**Comment by raga-adbe-gh** [comment]
Thanks @sukamat,
Suggested changes updated, test code for access token and driveid removed. Commented code removed

**Pull request merged**